cdap-security/src/main/java/co/cask/cdap/security/server/JAASLoginService.java

```diff
@@ -220,7 +227,14 @@ public void handle(Callback[] callbacks) throws IOException, UnsupportedCallback
       LoginContext loginContext = new LoginContext(loginModuleName, subject, callbackHandler, configuration);
 
       loginContext.login();
-
+      
+      String userNameCH = getUserName(callbackHandler);
+      if(ldap_caseconversion !=null && ldap_caseconversion.equalsIgnoreCase("lower")){
+        userNameCH = userNameCH.toLowerCase();
+      } else if (ldap_caseconversion !=null && ldap_caseconversion.equalsIgnoreCase("upper")){
+        userNameCH = userNameCH.toUpperCase();
+      }
+      
```

Comment:
- Variable 'usernameCH' is not being used while creating JAASUserPrincipal object. Please check.

- We should do similar changes for all types of authentication for eg. KnoxJAASLoginService. Not sure how this can be done for Basic Authentication. Did you consider changing grantToken API in GrantAccessToken.java ?

Reply:
1. missed it during copy paste
2. no, I just focused on LDAP authentication, let's discuss it